mips/uart_cbus.c

```diff
@@ -29,7 +31,7 @@ void uart_init() {
   LCR = LCR_8BITS; /* 8-bit data, no parity */
 }
 
-int uart_putchar(int c) {
+static void cbus_uart_putc(console_t *dev __unused, int c) {
```

Comment:
I'm interested where would buffering come into play once we decide to implement it. I suppose it would be up to the driver, so this function would store data into a synchronized cyclic buffer and return immediately, while another thread would wait until next byte can be written (presumably signaled by an interrupt), and read it from the buffer. Is my understanding correct?